xmind/tests/test_workbook_element.py

```diff
@@ -0,0 +1,751 @@
+
+        self.assertEqual(_test_el.TAG_NAME, TAG_WORKBOOK)
+        self.assertListEqual([
+            call((NAMESPACE, XMAP), NS_FO),
+            call((NAMESPACE, XMAP), NS_XHTML),
+            call((NAMESPACE, XMAP), NS_XLINK),
+            call((NAMESPACE, XMAP), NS_SVG)], self._setAttributeNS.call_args_list)
+        _getSheets.assert_called_once_with()
+        _createSheet.assert_called_once_with()
+        _addSheet.assert_called_once_with('sheet')
+
+    def test_WorkbookElement_init_called_with_diff_parameters(self):
+        _test_el_1 = WorkbookElement('node1')
+        self._workbook_mixin_element_init.assert_called_once_with('node1', None)
+
+        _test_el_2 = WorkbookElement(ownerWorkbook='workbook2')
+        self._workbook_mixin_element_init.assert_called_with(None, 'workbook2')
+
+        _test_el_3 = WorkbookElement('node3', 'workbook3')
+        self._workbook_mixin_element_init.assert_called_with('node3', 'workbook3')
+
+        self.assertEqual(self._workbook_mixin_element_init.call_count, 3)
+
+    def test_WorkbookElement_setOwnerWorkbook(self):
+        _test_el = WorkbookElement()
+
+        with self.assertRaises(Exception) as _ex:
+            _test_el.setOwnerWorkbook('ownerbook')
+
+        self.assertTrue(_ex.exception.args[0].find("WorkbookDocument allowed only contains one WorkbookElement") == 0)
+
+    def test_WorkbookElement_getSheet_sheets_is_none(self):
+        _test_el = WorkbookElement()
+        self._remove_patched_function('_getChildNodesByTagName')
+        self._remove_patched_function('_getOwnerWorkbook')
+
+        _getChildNodesByTagName = self._init_patch_with_name(
+            '_getChildNodesByTagName', 'xmind.core.Node.getChildNodesByTagName', return_value=None)
+        _getOwnerWorkbook = self._init_patch_with_name(
+            '_getOwnerWorkbook', 'xmind.core.mixin.WorkbookMixinElement.getOwnerWorkbook',
+            return_value=self._owner_workbook)
+
+        with self.assertRaises(Exception) as _ex:
+            _test_el.getSheets()
+
+        _getChildNodesByTagName.assert_called_once_with(TAG_SHEET)
+        _getOwnerWorkbook.assert_called_once_with()
+        self.assertTrue(_ex.exception.args[0].find("'NoneType' object is not iterable") != -1)
+
+    def test_WorkbookElement_getSheet_is_sheets_ownerbook_none(self):
+        _test_el = WorkbookElement()
+        self._remove_patched_function('_getChildNodesByTagName')
+        self._remove_patched_function('_getOwnerWorkbook')
+        self._remove_patched_function('_sheet_element_init')
+
+        _getChildNodesByTagName = self._init_patch_with_name(
+            '_getChildNodesByTagName', 'xmind.core.Node.getChildNodesByTagName', return_value=[11, 12, 13])
+        _getOwnerWorkbook = self._init_patch_with_name(
+            '_getOwnerWorkbook', 'xmind.core.mixin.WorkbookMixinElement.getOwnerWorkbook',
+            return_value=None)
+
+        _patcher = patch('xmind.core.workbook.SheetElement')
```

Comment:
To avoid holding of _patcher you could use ``with``` construction. It will do this job for you.

Reply:
Done